### PR TITLE
Change mutiple code snippets execution procedure

### LINKED ISSUE #35685: [Shell] Produce incorrect output
**Description:**
With following BBE Ballerina Shell produce wrong results(first case) , But it works fine if snippets are separately invoked (second case)
https://ballerina.io/learn/by-example/isolated-objects.html 
image

**Steps to reproduce:**
```bal
=$                                                               
 > 
 > import ballerina/io;
 > 
 > isolated class Counter {
 > 
 >     private int n = 0;
 > 
 >     isolated function get() returns int {
 >         lock {
 > 
 >             return self.n;
 > 
 >         }
 >     }
 > 
 >     isolated function inc() {
 >         lock {
 >             self.n += 1;
 >         }
 >     }
 > }
```

First case
```bal
=$  Counter c = new;
 > 
 >  c.inc();
 >  int v = c.get();
 >  io:println(v);
```
Second case
```bal
=$  Counter c = new;
 > 
 >  c.inc();

=$  int v = c.get();
 >  io:println(v);
 ```

### PR DESCRIPTION
**Pull request by DamithaSenevirathne**
## Purpose

Fixes #35685

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/BallerinaShell.java [hunk 1]
```diff
@@ -118,30 +116,27 @@ public void run() {
...
                             }
-                        }
-                    } else if (shellCompilation.getExceptionStatus() == ExceptionStatus.INVOKER_FAILED) {
+                    } else if (shellReturnValue.get().getExceptionStatus() == ExceptionStatus.INVOKER_FAILED) {
                         if (isContainsUndefinedModules(evaluator.diagnostics())) {
                             currentStateDiagnostics = List.copyOf(evaluator.diagnostics());
                             evaluator.resetDiagnostics();
                             executeChanges(source, currentStateDiagnostics);
                         }
                     }
```
Comment (dulajdilshan) [inline ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/BallerinaShell.java:137]: Can't we improve this if-else by getting `shellReturnValue.get()` into a variable and using early return if `shellReturnValue.isPresent` is false?

File: ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/BallerinaShell.java [hunk 2]
```diff
@@ -382,10 +375,8 @@ private void executeChanges(String source, Collection<Diagnostic> diagnostics) {
             importModules(moduleImporter, modules);
             try {
                 terminal.println("");
-                ShellCompilation shellCompilation = evaluator.getCompilation(source);
-                Optional<PackageCompilation> compilation = shellCompilation.getPackageCompilation();
-                ShellReturnValue shellReturnValue = evaluator.getValue(compilation).get();
-                String result = shellReturnValue.getResult();
+                Optional<ShellReturnValue> shellReturnValue = evaluator.execute(source);
+                String result = shellReturnValue.get().getResult();
```
Comment (dulajdilshan) [inline ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/BallerinaShell.java:379]: What if we get ` null ` here?  (shellReturnValue.get())

File: ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/ClassLoadInvoker.java [hunk 3]
```diff
@@ -265,6 +265,65 @@ public PackageCompilation getCompilation(Collection<Snippet> newSnippets) throws
...
+            variableDeclarations.put(varDclnSnippet, varDclnSnippet.names());
+
+        } else if (newSnippet instanceof ModuleMemberDeclarationSnippet) {
+            ModuleMemberDeclarationSnippet moduleDclnSnippet = (ModuleMemberDeclarationSnippet) newSnippet;
+            Identifier moduleDeclarationName = moduleDclnSnippet.name();
+            moduleDeclarations.put(moduleDeclarationName, moduleDclnSnippet);
+            Set<Identifier> usedPrefixes = newSnippet.usedImports().stream()
+                    .map(Identifier::new).collect(Collectors.toSet());
+            newImports.put(moduleDeclarationName, usedPrefixes);
+
```
Comment (dulajdilshan) [inline ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/ClassLoadInvoker.java:298]: Here, as well

File: ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/BallerinaShell.java (hunk 1 above)
Comment (DamithaSenevirathne) [reply ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/BallerinaShell.java]: Added in ea054445b72a422651c331b20cc9c2bb6056a8f8

File: ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/ClassLoadInvoker.java [hunk 4]
```diff
@@ -265,6 +265,65 @@ public PackageCompilation getCompilation(Collection<Snippet> newSnippets) throws
...
+        variableNames = new ArrayList<>();
+
+        if (newSnippet instanceof ImportDeclarationSnippet) {
+            processImport((ImportDeclarationSnippet) newSnippet);
+
+        } else if (newSnippet instanceof VariableDeclarationSnippet) {
+            VariableDeclarationSnippet varDclnSnippet = (VariableDeclarationSnippet) newSnippet;
+            variableNames.addAll(varDclnSnippet.names());
+            variableDeclarations.put(varDclnSnippet, varDclnSnippet.names());
+
```
[the remark this answers is not in the document]
Comment (DamithaSenevirathne) [reply ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/ClassLoadInvoker.java]: Removed in ea054445b72a422651c331b20cc9c2bb6056a8f8

File: ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/ClassLoadInvoker.java (hunk 3 above)
Comment (DamithaSenevirathne) [reply ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/ClassLoadInvoker.java]: Removed in ea054445b72a422651c331b20cc9c2bb6056a8f8

**Pull request closed without merging**